cmake/Helpers.cmake

```diff
@@ -0,0 +1,790 @@
+			)
+
+			# 3rd party DLLs are somewhere in the build directory
+			IF (MSVC)
+				ADD_TEST(NAME Test_${plugin_name}
+					COMMAND ${CMAKE_COMMAND} -E env "PATH=${CMAKE_BINARY_DIR}/gemrb;$ENV{PATH}" $<TARGET_FILE_DIR:gemrb_core>/$<TARGET_FILE_NAME:Test_${plugin_name}>
+					WORKING_DIRECTORY $<TARGET_FILE_DIR:gemrb_core>
+				)
+			ELSE ()
+				ADD_TEST(NAME Test_${plugin_name}
+					COMMAND $<TARGET_FILE_DIR:gemrb_core>/$<TARGET_FILE_NAME:Test_${plugin_name}>
+					WORKING_DIRECTORY $<TARGET_FILE_DIR:gemrb_core>
+				)
+			ENDIF()
+		# set env to where the core lib is
+		ELSE()
+			ADD_TEST(NAME Test_${plugin_name}
+				COMMAND $<TARGET_FILE:Test_${plugin_name}>
+				WORKING_DIRECTORY $<TARGET_FILE_DIR:gemrb_core>
+			)
+			SET_PROPERTY(TEST Test_${plugin_name} PROPERTY ENVIRONMENT "LD_LIBRARY_PATH=$<TARGET_FILE_DIR:gemrb_core>")
+			SET_PROPERTY(TEST Test_${plugin_name} PROPERTY ENVIRONMENT "DYLD_LIBRARY_PATH=$<TARGET_FILE_DIR:gemrb_core>")
+		ENDIF()
+	ENDIF()
+ENDFUNCTION()
+
+# pretty-print options macro
+# as of 2.8 cmake does not support anything like EVAL
+MACRO(PRINT_OPTION option)
+if (${option})
+	message(STATUS "  ${option}: ${${option}}")
+else()
+	message(STATUS "  ${option}: disabled")
+endif()
+ENDMACRO(PRINT_OPTION)
+
+FUNCTION(CONFIGURE_APPLE_SPECIFICS)
+    IF(APPLE)
+        # favor mac frameworks over unix libraries
+        SET(CMAKE_FIND_FRAMEWORK FIRST)
+        SET(BAKE_ICON 0)
+        set(CMAKE_OSX_DEPLOYMENT_TARGET "10.9" CACHE STRING "Minimum OS X deployment version")
+        # Make sure we can find the 'ibtool' program. we need it to compile xibs
+        FIND_PROGRAM(IBTOOL ibtool HINTS "/usr/bin" "${OSX_DEVELOPER_ROOT}/usr/bin")
+        IF (${IBTOOL} STREQUAL "IBTOOL-NOTFOUND")
+            MESSAGE (FATAL_ERROR "ibtool can not be found and is needed to compile the .xib files. It should have been installed with 
+                        the Apple developer tools. The default system paths were searched in addition to ${OSX_DEVELOPER_ROOT}/usr/bin")
+        ENDIF ()
+
+        ADD_DEFINITIONS("-x objective-c++")
+        INCLUDE_DIRECTORIES(platforms/apple)
+        FIND_LIBRARY(COCOA_LIBRARY_PATH Cocoa)
+        FIND_LIBRARY(COREFOUNDATION_LIBRARY CoreFoundation)
+        # for objective-c++
+        STRING(APPEND CMAKE_CXX_FLAGS " -DTARGET_OS_MAC")
+        # for pure objective-c
+        STRING(APPEND CMAKE_C_FLAGS " -x objective-c -DTARGET_OS_MAC")
+    ELSE(APPLE)
+        SET(BAKE_ICON 1)
+    ENDIF()
+    SET(BAKE_ICON ${BAKE_ICON} CACHE INTERNAL "")
+ENDFUNCTION()
+
+FUNCTION(CONFIGURE_VITA_SPECIFICS)
+    IF(VITA)
+        INCLUDE(platforms/vita/vita.cmake)
+    ENDIF(VITA)
+ENDFUNCTION()
+
+FUNCTION(CONFIGURE_UNIX_SPECIFICS)
+    IF(UNIX)
+        SET(CMAKE_THREAD_PREFER_PTHREAD true)
+        SET(CMAKE_THREAD_PREFER_PTHREAD ${CMAKE_THREAD_PREFER_PTHREAD} CACHE INTERNAL "")
+        FIND_PACKAGE(Threads REQUIRED)
+
+        # TODO ONLY FOR UBUNTU
```

Comment:
This new comment should be removed. If it bothers you, you can  remove the Ubuntu mention in the line below.